src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -9991,72 +9990,3081 @@ static void Main()
+                    Diagnostic(ErrorCode.ERR_CollectionExpressionEscape, "[x, y, z]").WithArguments("MyCollection<T>").WithLocation(12, 60));
+            }
+            else
+            {
+                var verifier = CompileAndVerify(comp,
+                    verify: Verification.Skipped,
+                    expectedOutput: IncludeExpectedOutput("[], [1, 2, 3], "));
+                verifier.VerifyIL("Program.Empty<T>", """
+                    {
+                      // Code size       16 (0x10)
+                      .maxstack  1
+                      IL_0000:  call       "T[] System.Array.Empty<T>()"
+                      IL_0005:  newobj     "System.ReadOnlySpan<T>..ctor(T[])"
+                      IL_000a:  call       "MyCollection<T> MyCollectionBuilder.Create<T>(scoped System.ReadOnlySpan<T>)"
+                      IL_000f:  ret
+                    }
+                    """);
+                if (targetFramework == TargetFramework.Net80)
+                {
+                    verifier.VerifyIL("Program.ThreeItems<T>", """
+                        {
+                          // Code size       64 (0x40)
+                          .maxstack  2
+                          .locals init (<>y__InlineArray3<T> V_0)
+                          IL_0000:  ldloca.s   V_0
+                          IL_0002:  initobj    "<>y__InlineArray3<T>"
+                          IL_0008:  ldloca.s   V_0
+                          IL_000a:  ldc.i4.0
+                          IL_000b:  call       "InlineArrayElementRef<<>y__InlineArray3<T>, T>(ref <>y__InlineArray3<T>, int)"
+                          IL_0010:  ldarg.0
+                          IL_0011:  stobj      "T"
+                          IL_0016:  ldloca.s   V_0
+                          IL_0018:  ldc.i4.1
+                          IL_0019:  call       "InlineArrayElementRef<<>y__InlineArray3<T>, T>(ref <>y__InlineArray3<T>, int)"
+                          IL_001e:  ldarg.1
+                          IL_001f:  stobj      "T"
+                          IL_0024:  ldloca.s   V_0
+                          IL_0026:  ldc.i4.2
+                          IL_0027:  call       "InlineArrayElementRef<<>y__InlineArray3<T>, T>(ref <>y__InlineArray3<T>, int)"
+                          IL_002c:  ldarg.2
+                          IL_002d:  stobj      "T"
+                          IL_0032:  ldloca.s   V_0
+                          IL_0034:  ldc.i4.3
+                          IL_0035:  call       "InlineArrayAsReadOnlySpan<<>y__InlineArray3<T>, T>(in <>y__InlineArray3<T>, int)"
+                          IL_003a:  call       "MyCollection<T> MyCollectionBuilder.Create<T>(scoped System.ReadOnlySpan<T>)"
+                          IL_003f:  ret
+                        }
+                        """);
+                }
+                else
+                {
+                    verifier.VerifyIL("Program.ThreeItems<T>", """
```

Comment:
do we have a tracking issue for adding an "unexpected heap alloc" warning here?

Reply:
https://github.com/dotnet/roslyn/issues/69693